### DISCUSSION TITLE
network plugin fails to install

### DISCUSSION THREAD

**Discussion by hliebscher** [OrcaSlicer/OrcaSlicer#33] [General]
After Upgrade to 1.4 beta  (incl the last merge)

The network plugin fails to install when opening the program. 

It worked OK with v1.3.4 and the official Bambu Studio 1.4
at MacOS 12.1

deleting the plugins directory is the same ...

**Comment by SoftFever** [comment]
Does it fix the problem if you rename(don't delete it contains all your configs) the `/Users/<yourname>/Library/Application Support/BambuStudio-SoftFever` folder?

**Comment by SoftFever** [comment]
It works fine on my Mac(12.6.1)

**Comment by hliebscher** [comment]
i found the bug

[download_plugin]: get version of settings's type, "slicer/plugins/cloud"
2022-12-18 11:50:36.409071[Thread 0x000000029a507000]:[download_plugin 1]: get type slicer/plugins/cloud, version 1.4.0.15, url https://public-cdn.bambulab.com/upgrade/studio/plugins/01.04.00.15/mac_01.04.00.15.zip
2022-12-18 11:50:36.409140[Thread 0x000000029a507000]:UpgradeNetworkJob: percent = 5msg = Downloading
2022-12-18 11:50:36.409371[Thread 0x000000029a507000]:[download_plugin] get_url = https://public-cdn.bambulab.com/upgrade/studio/plugins/01.04.00.15/mac_01.04.00.15.zip
2022-12-18 11:50:36.983294[Thread 0x000000029a507000]:UpgradeNetworkJob: percent = 0msg = Download failed
2022-12-18 11:50:36.983460[Thread 0x000000029a507000]:[download_plugin 2] on_error: curl:**SSL connect error:
SSL peer handshake failed, the server most likely requires a client certificate to connect**
[Error 35], body = 


manually download works

https://public-cdn.bambulab.com/upgrade/studio/plugins/01.04.00.15/mac_01.04.00.15.zip

**Reply by Manfredo1952** [reply]
I am on OSX 10.7.2 - Network plugins fails to load. Downloaded manually, copied both files to plugins folder.  Didnt work first try, had to use the " xattr -d com.apple.quarantine /Users/username/Library/Application\ Support/BambuStudio-SoftFever/plugins/libBambuSource.dylib " on both files. Thereafter it does work now - thanks for the hint re manual download

**Comment by SoftFever** [comment]
Cool.
Thanks for sharing the investigation.
I'm closing this issue for now.
We can revisit it if there are more reports in the future.

-SoftFever

**Comment by GP3D84** [comment]
Hi. 
I have the same bug on macOS Ventura with latest 1.4 beta 3
```* * Connection #0 to host api.bambulab.com left intact
*   Trying 99.86.91.19:443...
* Connected to public-cdn.bambulab.com (99.86.91.19) port 443 (#0)
* ALPN, offering http/1.1
* WARNING: disabling hostname validation also disables SNI.
* SSL peer handshake failed, the server most likely requires a client certificate to connect
* Closing connection 0
```

**Comment by HakunMatat4** [comment]
This isn't a SF problem.
On Linux, I can download the file but there is a problem with the file itself.
This is a BL problem to deal with.

`BambuStudio-SoftFever/plugins/libbambu_networking.so: undefined symbol: __libc_single_threaded`

**Comment by hliebscher** [comment]
It works after another compile with mac without any problems...
it can be a problem of the Loadbalancer from CDN from Bambulab

**Reply by HakunMatat4** [reply]
This was not a problem before, only after BambuStudio update. Even running the latest version, it at least install it but keep doing this every time you open the slicer.

**Reply by hliebscher** [reply]
win or MAC Version ?

**Reply by hliebscher** [reply]
*   Trying 54.215.111.72:443...
* Connected to api.bambulab.com (54.215.111.72) port 443 (#0)
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
*   Trying 50.18.154.192:443...
* Connected to e.bambulab.com (50.18.154.192) port 443 (#0)
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=*.bambulab.com
*  start date: Jun 30 00:00:00 2022 GMT
*  expire date: Jul 15 23:59:59 2023 GMT
*  issuer: C=US; O=DigiCert, Inc.; CN=RapidSSL Global TLS RSA4096 SHA256 2022 CA1
*  SSL certificate verify ok.
> GET /v1/iot-service/api/slicer/resource?

**Reply by HakunMatat4** [reply]
So myself among others have been trying to compile the Linux version of the slicer for ages, long story short BL said on that ticket that glibc was updated.

The whole system depends on it, I'm pretty sure MacOS is the same.

The reason why this network plugin is going nuts on some Linux after the slicer release is because of it. Network plugin depends on newer glibc version.

You cannot just update glibc to its minimal required version which I believe it's 2.32 IIRC if your system doesn't have it installed, or you will open Pandora's box.

On my case, I need to upgrade my OS to its latest LTS release.

I have no idea how macOS works so yeah, the problem is lot deeper.

**Reply by hliebscher** [reply]
for witch linux version do you need BambuStudio ?

**Reply by HakunMatat4** [reply]
> for witch linux version do you need BambuStudio ? 

I have all the versions, they merged a PR that fixed all the problems BUT the latest release seems to use latest OS features so a bunch of us on running system that weren't released recently cannot see the plate hahaha

Yep, the slicer opens and all but there's no plate.
